src/structures/paging/recursive.rs

```diff
@@ -1,21 +1,51 @@
 use registers::control::Cr3;
+use instructions::tlb;
 use structures::paging::page_table::{FrameError, PageTable, PageTableEntry, PageTableFlags};
 use structures::paging::{NotGiantPageSize, Page, PageSize, PhysFrame, Size1GB, Size2MB, Size4KB};
 use ux::u9;
 use VirtAddr;
 
+use core::mem;
+
+/// This type must be used and will either flush the modified page or can be unsafely ignored.
+#[must_use = "Page Table changes must be flushed or unsafely ignored."]
+pub struct MapperFlush<S: PageSize>(Page<S>);
+
+impl<S: PageSize> MapperFlush<S> {
+    /// Create a new flush promise
+    fn new(page: Page<S>) -> Self {
+        MapperFlush(page)
+    }
+
+    // Flush 
+    pub fn flush(self) {
+        tlb::flush(self.0.start_address());
+        mem::forget(self);
+    }
+
+    pub unsafe fn ignore(self) {
+        mem::forget(self);
+    }
+}
+
+impl<S: PageSize> Drop for MapperFlush<S> {
+    fn drop(&mut self) {
+        panic!("`MapperFlush` must be used.")
```

Comment:
Hmm, not sure if we should really panic on drop. I think the warning through the must_use attribute is enough, because nothing bad happens when you forget it (you need to explicitly access the address through an unsafe block later).

Reply:
Okay, I'll remove it!